org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/embedder/LifecycleManagerDisposer.java

```diff
@@ -61,13 +61,7 @@ private void disposeOutdatedEntries(ClassRealm realm, LifecycleManager manager)
     lifecyclesMapField.setAccessible(true);
     Object lifecycles = lifecyclesMapField.get(manager);
     if(lifecycles instanceof Map<?, ?> map) {
-      for(Object key : map.keySet().toArray()) {
-        if(key instanceof Class<?> clazz) {
-          if(clazz.getClassLoader() == realm) {
-            map.remove(key);
-          }
-        }
-      }
+      map.keySet().removeIf(key -> key instanceof Class<?> clazz && clazz.getClassLoader() == realm);
```

Comment:
@laeubi that's what I had in mind in https://github.com/eclipse-m2e/m2e-core/pull/1217#discussion_r1084948482. :)